webClient/src/app/app.component.ts

```diff
@@ -182,6 +183,7 @@ export class AppComponent implements AfterViewInit {
         }
         this.host = config.contents.host;
         this.port = config.contents.port;
+        this.setSavedSettings();
```

Comment:
By doing this after the above `this.loadConfig().subscribe`, it actually makes 2 network requests for an object of the same type.
The config service (which is where the call is doing) supports a hierarchy: if your user doesnt have a file, it can find a file of the same name which is a default.
So in this case, it is only necessary to do 1 network call: the first one will return the contents of your users preference if it exists.
More on this in the next comment.